### PR TITLE
Another stab at #673

### PR DESCRIPTION
**Pull request by Lapz**
Another stab at #673 but this time much simpler

* Added rustfmt to the CI.
* Ran rustfmt on the whole codebase

**Note**: Running cargo fmt will error on stable but a fix is present in nightly

### CODE REVIEW

**Comment by cmyr** [comment]
Can you clarify what you mean about this erroring on stable? it seems to work for me?

File: rust/core-lib/src/client.rs [hunk 1]
```diff
@@ -42,150 +42,176 @@ impl Client {
...
 
-    pub fn update_cmds(&self, view_id: ViewId,
-                       plugin: &str, cmds: &[Command]) {
-        self.0.send_rpc_notification("update_cmds",
-                                     &json!({
+    pub fn update_cmds(&self, view_id: ViewId, plugin: &str, cmds: &[Command]) {
+        self.0.send_rpc_notification(
+            "update_cmds",
+            &json!({
                                          "view_id": view_id,
```
Comment (cmyr) [inline rust/core-lib/src/client.rs]: it looks like maybe rustfmt is leaving the contents of the macro alone or something? This alignment seems screwy

**Comment by cmyr** [comment]
1. We'll also want to update the `rust/run_all_checks` script to run `rustfmt`.

2. I think maybe this could be good motivation for me to finish up some outstanding reviews, so that we can get any larger PRs merged before running this, because merging after or rebasing might be hairy.

In any case will try and get this resolved quickly.

Thanks @Lapz!

File: rust/core-lib/src/client.rs (hunk 1 above)
Comment (Lapz) [reply rust/core-lib/src/client.rs]: Rust fmt dosen't touch macros so it will have to be done manually

**Comment by Lapz** [comment]
I've you changes in #850 in as well and made the change to run_all_checks script

**Comment by cmyr** [comment]
Okay, thanks for grabbing the stuff from 850. We'll also want to rerun `cargo fmt` now that we're ignoring those files.

I've pushed #851, which I'm going to merge shortly, and which will make everything work on stable for now.

My suggestion:

- once #851 is merged, fetch master
- `git rebase -i master`, dropping 957ae88
- rerun `cargo fmt`. You might need to run it twice
- add that commit here
- if we get through CI, we can merge.

One other note is that rustfmt _might_ produce different results on 1.28 (which we use in CI) and 1.29 (current stable). You can install 1.28 through rustup with `rustup toolchain add 1.28.0`, and then run `rustup component add rustfmt-preview`.

File: rustfmt.toml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+force_explicit_abi = true
+max_width = 120
```
Comment (cmyr) [inline rustfmt.toml]: Can we get rid of this line? I think `100` is fine, and our original motivation for 120 (do what fuchsia does) is less relevant now.

**Comment by cmyr** [comment]
@Lapz So I think this probably requires us to move a bit more quickly than your schedule might allow, so with your approval I would like to:

- cherry-pick your commits here that involve updating the travis config, adding rustfmt.toml, and the docs
- commit these changes, without any rustfmt changes


This will then let us commit the actual reformatted files separately, in chunks, taking care not to create any big merge conflicts for active PRs.

In any case I appreciate the work you've put into this!

**Comment by cmyr** [comment]
@Lapz I think this is too big a change to try and do all at once, and I think it was probably unfair of me to ask you to do it; it was always going to be hard to get this merged. I'm going to take this on myself, and do it in stages, trying to avoid interfering with other work. Thank you for the effort, and I'd be happy to find another better starter issue for you if you like!

**Pull request closed without merging**